roomserver/internal/perform/perform_create_room.go

```diff
@@ -63,13 +63,12 @@ func (c *Creator) PerformCreateRoom(ctx context.Context, userID spec.UserID, roo
 			}
 		}
 	}
-	senderID, err := c.DB.GetSenderIDForUser(ctx, roomID.String(), userID)
-	if err != nil {
-		util.GetLogger(ctx).WithError(err).Error("Failed getting senderID for user")
-		return "", &util.JSONResponse{
-			Code: http.StatusInternalServerError,
-			JSON: spec.InternalServerError{},
-		}
+	var senderID spec.SenderID
+	if createRequest.RoomVersion == gomatrixserverlib.RoomVersionPseudoIDs {
+		// TODO: pseudoIDs - generate senderID kere!
```

Comment:
From the looks of it, we can't do it that early, as we don't have a roomNID yet. ([see #3108](https://github.com/matrix-org/dendrite/pull/3108/files#diff-4005796ff792149af3a30f8c6a020b13ca1a33e37b7ccb6db645fd805449f9a2R357-R380))
What we could do, I guess, create bypass all checks and "reserve" a roomNID for the roomID.